### ISSUE TITLE
[BUG] highlight keyword does not affect padded content

### ISSUE THREAD

**Issue by rybesh** [willmcgugan/rich#1404]
**Describe the bug**

The `highlight` keyword does not affect padded text, either to turn highlighting off or to turn it on.

**To Reproduce**

```python
from rich.console import Console
from rich.padding import Padding
console = Console(highlight=True)
console.print(Padding("1 2 3 4", (1,1)), highlight=False)
```

Expected output: numbers are not highlighted.
Actual output: numbers *are* highlighted.

```python
from rich.console import Console
from rich.padding import Padding
console = Console(highlight=False)
console.print(Padding("1 2 3 4", (1,1)), highlight=True)
```

Expected output: numbers are highlighted.
Actual output: numbers are *not* highlighted.

**Platform**

macOS 11.5.1
iTerm2 3.4.8
rich==10.7.0

**Comment by mlittlec** [comment]
Reproduced the behaviour on Windows 10 (and Windows 11 Beta) 

**Platform**

MS Windows 10 (and 11 Beta)
Windows Terminal 1.9.1942.0
rich==10.0.7.0

_Possible explanation_
It appears that setting this line:

   console = Console(highlight=False)

overrides the use of the 'highlight=True' in this line:

  console.print(Padding("1 2 3 4", (1,1)), highlight=True)

It unclear to me at this time whether there is any hierarchy controlling whether it is possible to override the systemwide setting of 'highlight' variable in the Console() function or not.

**Comment by willmcgugan** [comment]
Should be fixed in v10.8.0
